lib/arqo/query.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module ARQO
+  # Parent class for query objects
+  class Query
+    attr_reader :relation
+    delegate_missing_to :relation
+
+    def initialize(relation = derived_relation)
+      @relation = relation
+      @relation = @relation.extending(scope_module) if Object.const_defined?(scope_module_name)
```

Comment:
Also maybe (more of a question) what do you think if we freeze the relation object? so that no query method is allowed to mutate it (in any weird case scenario where the end user might want to reassign the `@relation`)

Reply:
good idea 💯

Reply:
@juanmanuelramallo seems like this is not possible, that would prevent users to apply more things to the `ActiveRecord::Relation`.

![image](https://user-images.githubusercontent.com/6373536/85319159-86a14a00-b497-11ea-806a-c08368fc9249.png)